Explainers/Render_Blocking_Status.md

```diff
@@ -0,0 +1,159 @@
+# Render Blocking Status
+
+Proposal to add a new field `render blocking status` to PerformanceResourceTiming which is an [enum](https://webidl.spec.whatwg.org/#idl-enums)  highlighting the status of stylesheets and scripts.
+
+## Use cases
+
+RUM providers could then easily determine the render blocking resources without having to rely on complex heurestics to determine the same.
+This would enable analysis for below scenaarios and possibly more.
+- Determine resources downloaded before FCP but were not render blocking? (and hence, could've been delayed)
+- Determine resources that were render blocking, but weren't discovered early enough? (and hence, could've benefited from being preloaded)
+(Reference : https://github.com/w3c/resource-timing/issues/262)
+
+
+## API Changes and Example Code
+
+The PerformanceResourceTiming Interface in <a href="https://w3c.github.io/resource-timing/#sec-performanceresourcetiming">resource-timing</a> would be updated to 
+```bash
+[Exposed=(Window,Worker)]
+interface PerformanceResourceTiming : PerformanceEntry {
+    ...
+    ...
+    readonly attribute RenderBlockingStatusType renderBlockingStatus;
+    ...
+    ...
+    [Default] object toJSON();
+};
+
+enum RenderBlockingStatusType {
+    "blocking",
+    "non-blocking"
+};
+```
+
+Sample usage:
+```javascript
+const entry_list = performance.getEntriesByType("resource");
+console.log(entry_list[0].renderBlockingStatus);
+```
+
+
+## Render Blocking Status Values
+
+The following values are proposed for the resource blocking status enum
+
+`blocking` - A potentially render blocking resource
+
+`non-blocking` - Non blocking resource
+
+We primarily reuse the value of the [render blocking](https://fetch.spec.whatwg.org/#request-render-blocking) boolean associated with fetch [request](https://fetch.spec.whatwg.org/#concept-request)
```

Comment:
```suggestion
We primarily reuse the value of the [render-blocking](https://fetch.spec.whatwg.org/#request-render-blocking) boolean associated with fetch [request](https://fetch.spec.whatwg.org/#concept-request)
```

Also the [HTML spec](https://html.spec.whatwg.org/#render-blocking-mechanism) uses "render-blocking" rather than "render blocking", even when not explicitly referring to the `render-blocking` boolean, which seems more correct to me. Should we use that format throughout this doc (which would also avoid errors like this when actually refer to the boolean)?

And, if so, possibly rename the file too to `Render-blocking_Status.md` - or would that mix of dashes and underscores just be confusing?

Reply:
Thanks. I've made the changes everywhere in the doc. It does make sense to allign with the existing terminology in the spec. 

I also think now I should update it in the spec PR (https://github.com/w3c/resource-timing/pull/327) as well, so that that is also in line with the existing terminology. WDYT?

> And, if so, possibly rename the file too to Render-blocking_Status.md - or would that mix of dashes and underscores just be confusing?

This explainer in my repo was linked to various places, so if I change the name those links might become inaccessible. I'll rename it in a follow up PR after this lands?

Reply:
Thanks for that. Updating the spec PR similarly seems sensible to me.